x/pointer/keeper/msg_server_add_erc_20_native_pointer.go

```diff
@@ -0,0 +1,48 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/vm"
+
+	"github.com/titantkx/titan/x/pointer/types"
+)
+
+func (k msgServer) AddERC20NativePointer(goCtx context.Context, msg *types.MsgAddERC20NativePointer) (*types.MsgAddERC20NativePointerResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// validate the authority
+	authority, err := sdk.AccAddressFromBech32(msg.Authority)
+	if err != nil {
+		return nil, err
+	}
+	if authority.String() != k.authority {
+		return nil, types.WrapErrorf(sdkerrors.ErrUnauthorized, "incorrect authority %s", msg.Authority)
```

Comment:
```suggestion
		return nil, types.WrapError(sdkerrors.ErrUnauthorized, "only gov is allowed to perform this operation")
```

Use more detail message. `Unauthorized` and `incorrect authority` are redundant